apps/infra/api/routes/etl.py

```
@@ -0,0 +1,242 @@
+            f"Task {task_id}: Sync completed successfully, {len(transactions)} transactions processed"
+        )
+
+    except Exception as e:
+        logger.error(f"Task {task_id}: Error in sync task: {e}")
+
+
+@router.post("/sync", response_model=SyncStatusResponse)
+async def sync_transactions(
+    request: SyncTransactionsRequest,
+    background_tasks: BackgroundTasks,
+    catalog=Depends(get_catalog),
```

Ruff (0.11.9)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)